pkg/webreg/webreg.go

```diff
@@ -365,6 +366,253 @@ const templateDefinitions = `
+  <li><code>pipeline:src</code> clones the code under test <code>FROM pipeline:root</code></li>
+  <li><code>pipeline:bin</code> runs commands in the cloned repository to build artifacts <code>FROM pipeline:src</code></li>
+  <li><code>pipeline:test-bin</code> runs a separate set of commands in the cloned repository to build test artifacts <code>FROM pipeline:src</code></li>
+</ul>
+
+<code>ci-operator</code> configuration:
+{{ yamlSyntax (index . "ciOperatorPipelineConfig") }}
+
+<p>
+The content created with these OpenShift <code>Builds</code> is addressable
+in the <code>ci-operator</code> configuration simply with the tag. For instance,
+the <code>pipeline:bin</code> image can be referenced as <code>bin</code> when
+the content in that image is needed in derivative <code>Builds</code>. 
+</p>
+
+<h4 id="images"><a href="#images">Building Container Images</a></h4>
+
+<p>
+Once container images exist that contain the output artifacts for a repository,
+output container images may be built that make use of those artifacts. Commonly,
+the desired output container image will contain only the executables for a
+component and not any of the build-time dependencies. Furthermore, most teams
+will need to publish their output container images through the automated release
+pipeline, which requires that the images are built in Red Hat's production image
+build system, OSBS. In order to create an output container image without build-time
+dependencies in a manner which is compatible with OSBS, the simplest approach is a
+multi-stage <code>Dockerfile</code> build.
+</p>
+
+<p>
+The ubiquitous pattern for a multi-stage <code>Dockerfile</code> runs a compilation
+in a builder image and copies the output artifacts into a production output image base.
+For instance:
+<p>
+
+<code>Dockerfile</code>:
+{{ dockerfileSyntax (index . "multistageDockerfile") }}
+
+<p>
+While such a <code>Dockerfile</code> could simply be built by <code>ci-operator</code>,
+a number of optimizations can be configured to speed up the process, especially if
+multiple output images share artifacts. An output container image build is configured
+for <code>ci-operator</code> with the <code>images</code> stanza in the configuration.
+Any entry in the <code>images</code> stanza can be configured with native OpenShift
+<code>Builds</code> options; the full list can be viewed <a href="https://godoc.org/github.com/openshift/ci-tools/pkg/api#ProjectDirectoryImageBuildInputs">here.</a>
+In the following example, an output container image is built where the <code>builder</code>
+image is replaced with the image layers containing built artifacts in <code>pipeline:bin</code>
+and the output image base is replaced with the appropriate entry from <code>base_images</code>.
+<p>
+
+<code>ci-operator</code> configuration:
+{{ yamlSyntax (index . "ciOperatorImageConfig") }}
+
+<p>
+By making use of the previously compiled artifacts in the intermediate <code>pipeline:bin</code>
```

Comment:
i don't see where the inlined example references pipeline:bin.  it refs `base`, `test-bin`, and `cli`

Reply:
```yaml
inputs:
  bin:
    as:
```